### PR TITLE
feat: Add webshop link functionality

### PR DESCRIPTION
**Pull request by afrojezus**
Closes https://github.com/Dintero/Dintero.Discounts.Web.SDK/issues/71

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,222 +1,219 @@
...
+            position: absolute;
+            cursor: pointer;
+            top: 0;
+            left: 0;
+            right: 0;
+            bottom: 0;
+            background-color: #ccc;
+            -webkit-transition: 0.4s;
+            transition: 0.4s;
+            border-radius: 34px;
```
Comment (frodeaa) [inline index.html:66]: can you revert the reformat of the code?

File: index.html [hunk 2]
```diff
@@ -1,222 +1,219 @@
...
+                "reward": { "type": "discount_item_new_price", "value": 5000 },
+                "created_by": "DinteroTest",
+                "created_at": "2021-06-04T10:21:18.029Z",
+                "updated_at": "2021-06-04T10:21:18.029Z",
+                "receipt_text": "Ukens tilbud",
+                "type": "item",
+                "links": [
+                    {
+                        "rel": "medium_discount_image",
+                        "href": "https://uploads-ssl.webflow.com/5f023ba9549ac22870f32943/5f1333ae5a496b0e0476975e_dintero-logo.svg"
```
Comment (frodeaa) [inline index.html:195]: maybe add an webshop link to the example?

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,15343 @@
 {
     "name": "@dintero/discounts-web-sdk",
     "version": "0.0.0-development",
-    "lockfileVersion": 1,
+    "lockfileVersion": 2,
     "requires": true,
+    "packages": {
```
Comment (frodeaa) [inline package-lock.json:6]: can you revert the changes?

File: package.json [hunk 4]
```diff
@@ -12,6 +12,7 @@
     },
     "scripts": {
         "test": "karma start",
+        "dev": "preconstruct watch",
```
Comment (frodeaa) [inline package.json:15]:
```suggestion
        "watch": "preconstruct watch",
```

I think we usually calls the script for watch or start

File: src/types.d.ts [hunk 5]
```diff
@@ -581,6 +581,7 @@ export type Configuration = {
         limit?: number,
         discountId?: string,
     }
+    linkBehaviour?: 'new_tab' | 'direct',
```
Comment (frodeaa) [inline src/types.d.ts:584]:
```suggestion
    linkTarget?: 'blank' | '_self', 
```

File: src/index.ts [hunk 6]
```diff
@@ -8,9 +8,11 @@ import { normalize } from "./normalize";
 import pkg from "../package.json";
 
 
+
 const defaultConfig:Partial<Configuration> = {
     language: 'no',
     version: pkg.version,
+    linkBehaviour: 'direct',
```
Comment (frodeaa) [inline src/index.ts:15]:
```suggestion
    linkBehaviour: '_self',
```

File: src/index.ts [hunk 7]
```diff
@@ -99,6 +101,12 @@ const renderDeals = (configuration: Configuration, discounts: Discount[]):Embed
     const wrapper = createElement({ tag: "div", styles: [normalize, wrapperStyles], theme: configuration.theme });
     discounts.forEach((discount) => {
         const elem = createDiscount(discount, configuration);
+        if (containsWebShopLink(discount)) {
+            if (configuration.linkBehaviour && configuration.linkBehaviour === 'new_tab') {
+                elem.setAttribute('target', '_blank');
+            }
+            elem.setAttribute('href', discount.links.find(x => x.rel === 'webshop').href);
+        }
```
Comment (frodeaa) [inline src/index.ts:109]:
```suggestion
        const webShopLink = findWebshopLink(discount);
        if (webShopLink) {
            elem.setAttribute('target', configuration?.linkBehaviour || '_self');
            elem.setAttribute('href', webShopLink.href);
        }
```

maybe simplify it?

**Review by frodeaa — CHANGES_REQUESTED** [review]
Looks good, please revert formatting and package-lock.json

File: index.html (hunk 2 above)
Comment (afrojezus) [reply index.html]: added

**Review by svnv — APPROVED** [review]

**Review by frodeaa — APPROVED** [review]

**Comment by frodeaa** [comment]
Added "feat:" prefix to the PR title, it should then trigger a releas of the SDK when merged into master

@afrojezus @svnv https://github.com/Dintero/Dintero.Discounts.Web.SDK#creating-a-new-release
